dali/operators/reader/fits_reader_gpu_op.cu

```diff
@@ -0,0 +1,417 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/kernels/dynamic_scratchpad.h"
+#include "dali/operators/reader/fits_reader_gpu_op.h"
+#include "dali/pipeline/data/tensor_list.h"
+
+#include <string>
+#include <vector>
+
+namespace dali {
+
+__constant__ int nonzero_count[256] = {
+    0, 1, 2, 2, 3, 3, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,
+    6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6, 6,
+    7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7,
+    7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7, 7,
+    8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8,
+    8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8,
+    8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8,
+    8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8};
+
+__global__ void rice_decompress_byte(unsigned char *compressed_data,
+                                     unsigned char *uncompressed_data, const long *tile_offset,
+                                     const long *tile_size, int blocksize, long tiles,
+                                     long maxtilelen, double bscale, double bzero) {
+  int index = (int)(blockIdx.x * blockDim.x + threadIdx.x);
+  int stride = (int)(blockDim.x * gridDim.x);
+  for (long tile = index; tile < tiles; tile += stride) {
+    unsigned char *compressed_tile = compressed_data + tile_offset[tile];
+    int i, imax;
+    int k;
+    int nbits, nzero, fs;
+    unsigned int b, diff, lastpix;
+    int fsmax, fsbits, bbits;
+    long beg;
+
+    fsbits = 3;
+    fsmax = 6;
```

Comment:
Making them dependent would lead to some unpleasant calculations (especially for fsmax) so I moved them into arrays.
`const int fsbits[3] = {3, 4, 5};`
`const int fsmax[3] = {6, 14, 25};`